### ISSUE TITLE
BiglyBT Leaking when VPN goes down.

### ISSUE THREAD

**Issue by NaPurHab** [BiglySoftware/BiglyBT#830]
As per attached, I should always be bound to the VPN.  The VPN I use has Split-Tunneling.  This allows it to keep the main use on the default connection. But programs started via Split-Tunneling are under the VPN.  Biglybt is a special case, as it will not start via Split-Tunneling, but luckily it can see Purevpn as ppp1. So sees the Split-Tunneling interface anyway.

The problem is that if the VPN goes down, or the DSL Modem looses its connection (i.e. the ISP does overnight work that disconnects the Modem) and also takes down the VPN. Then Biglybt tries to work around it.  When the connection is lost, I will get up in the morning and find the VPN down. But worse, that Biglybt has somehow connected to the lo (Software Loopback Interface) 127.0.0.1 and is actually Uploading Data.

When the Enforce IP Binding even when not available is Set ON, this should not happen.

Recommend that when Binding is LOST, that a Pause All take effect, and Resume on restoration of the Binding. This would prevent Leaking, as everything would stop until the Binding is Restored. Usually by me waking up and finding the VPN Down.

It might also be a good idea to add a Entry Field under the VPN Detection, to enter the VPN Name. It could be used to Search for the VPN.

Hope to see an end the Leaking.

* Windows 10 1809
* 1.8.0.1_B01
* Java 1.8.0_201 (64 bit)
![biglybt advanced network settings](https://user-images.githubusercontent.com/37715552/51628286-a2a36380-1f12-11e9-8160-bdb2d512e761.PNG)

**Comment by parg** [comment]
How is it uploading over 127.0.0.1?

**Comment by parg** [comment]
Do you have I2P enabled? Perhaps check the peer connections under 'All Peers' to see what's connected?

**Comment by NaPurHab** [comment]
Yes I hae I2P enabled. At the moment I am using the system normally.  The ISP doing work, and knocking down the VPN only happens about once a month, sometime 2 times a month.  But when it happens, it is while I am sleeping.  They usually do the work between Midnight and 6am.

As to what is connected, Quite a lot. I have a 500/500mbit symmetrical connection. So I am seeding 60 uploads at the moment. But only get about 100 mbits avg speed. This is probably due to HD speed. So have spread it out over 6 HDs. If too many torrents are reading from one drive, I move things around to load balance.

But for testing purposes, I suggest connect to a VPN that sets a unique port name.  i.e. normal is an eth# and the VPN is a ppp#, Bind it, then kill the VPN and wait.

**Comment by parg** [comment]
The built-in I2P doesn't follow BiglyBT bindings - be interesting to see if your upload is over that. If you want more control over I2P then install I2P standalone and configure it how you want, then point BiglyBT at the 'external router'

**Comment by NaPurHab** [comment]
Still doesn't explain why when I mouse over the Routing, it shows 127.0.0.1.  Seems that the Enforce Bindings is not doing what it should.

**Comment by NaPurHab** [comment]
OK, just tested with I2P Helper off, did a restart Biglybt.   When I mouse over the Routing it says:
Bindings: 127.0.0.1, Force=yes
Missing: ppp1
No Connections

So far no torrents uploading over 127.0.0.1.  But still should it be Binding to 127.0.0.1.  Also I have the restart after idle set to 3 minutes. And with I2P not running, it is now restarting every 3 minutes. Seems I2P was the connection culprit.  I just paused all after a restart.  Ok, 3 minutes up and no auto restart. Seems pausing counts as activity, and stops the 3 minute idle restart.  So my idea of adding a Pause/Resume on loss of binding port would be a good idea.  Would allow I2P to be on, and port binding loss would not allow leaking if everything is paused.

So please consider adding Pause/Resume on Binding Loss.  Perhaps as an indented Sub-Item under Enforce Binding. That can only be selected if Enforce Binding is ON. And is greyed out if Enforce Binding is OFF.

**Comment by parg** [comment]
That's the way that it forces things not-to-work when the binding is down - it returns a 127.0.0.1 binding in the belief that this isn't going to route anywhere.

Problem with pause/resume is that there are many reasons that things might already be paused. The 'resume' on the VPN reconnecting will unpause everything. Of course you might be happy with this...

**Comment by NaPurHab** [comment]
Now all we need is a Wiki, BiglyBT needs its own.

**Comment by TuxPaper** [comment]
aww man, we missed our opportunity to hold your downloads hostage and demand $5.7b before releasing them! /s
